packages/synthetix-main/contracts/storage/Market.sol

```diff
@@ -21,6 +22,10 @@ library Market {
     using DecimalMath for uint256;
     using DecimalMath for int256;
     using DecimalMath for int128;
+    using SafeCast for uint128;
```

Comment:
thats a lot of `using`!

note: solidity does support function overloading btw, so rather than having function names in the lib like `int256TouInt128()`, it could just be `.toInt()` and the function takes in a `int128` parameter. This would make for much more readable casts.

Reply:
It does support it, but it is very buggy for similar types. We can improve on it later, raise issues to the Solidity team, etc.